### ISSUE TITLE
PretrainedTrasformerIndexer can't pad GPT2

### ISSUE THREAD

**Issue by nickswalker** [allenai/allennlp#4126]
**Describe the bug**
The PretrainedTransformerIndexer pulls the pad token from the configured pretrained model's tokenizer:
https://github.com/allenai/allennlp/blob/583ea41392d3c173b09bd9f0e2ee0ce2e6befc9e/allennlp/data/token_indexers/pretrained_transformer_indexer.py#L195

But the pretrained GPT2 models don't have padding, so this is `None`. The code crashes when it actually tries to pad a couple of lines later.

**To Reproduce**
Steps to reproduce the behavior
1. Use `gpt2` with a `pretrained_transformer` token indexer

**Expected behavior**
Not crashing, but beyond that not sure. I take it an arbitrary token could be used to pad given that the attention mask should ignore those indices anyways. If that's true then this could be implemented as a special case in the indexer.

**System (please complete the following information):**
 - OS: [e.g. OSX, Linux]
 - Python version: 3.6.9
 - AllenNLP version: I installed from master
 - PyTorch version: 1.4.0

**Additional context**
The lack of a pad token has been discussed here https://github.com/huggingface/transformers/issues/2630

**Comment by dirkgr** [comment]
Looks like we can just fill in a `0` in this case.

**Comment by dirkgr** [comment]
Actually, this is pretty stupid. Having both a mask and a padding token is redundant, yes. But the proper solution is to always have a padding token and get rid of the mask, not the other way around. With the mask, we have to carry around another tensor for no reason, and make sure everywhere that it matches the shape of the other tensor.
